src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,93 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+
 public class WorkWithFile {
+    private static final String BUY_OPERATION = "buy";
+    private static final String SUPPLY_OPERATION = "supply";
+    private static final String RESULT_OPERATION = "result";
+    private static final String IO_OPERATION_ERROR = "I/O operation interrupted";
+    private static final String VALUES_SEPARATOR = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String readFromFileResult = readFromFile(fromFileName);
+        String fileContentProcessed = processFileReadResults(readFromFileResult);
+        writeToFile(fileContentProcessed, toFileName);
+    }
+
+    private static String readFromFile(String fromFileName) {
+        StringBuilder fileRead = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String fileReadLine = bufferedReader.readLine();
+            while (fileReadLine != null) {
+                fileRead.append(fileReadLine).append(System.lineSeparator());
+                fileReadLine = bufferedReader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File not found.", e);
+        } catch (IOException e) {
+            throw new RuntimeException(IO_OPERATION_ERROR, e);
+        }
+        return String.valueOf(fileRead);
+    }
+
+    private static String processFileReadResults(String fileContent) {
+        int supplyOperationSum = 0;
+        int buyOperationSum = 0;
+        try (BufferedReader bufferedReader = new BufferedReader(new StringReader(fileContent))) {
+            final int operationAmountIndex = 1;
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                String[] lineSplit = line.split(VALUES_SEPARATOR);
+                String operationType = lineSplit[0];
```

Comment:
```suggestion
                String operationType = lineSplit[OPERATION_INDEX];
```
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/234a6471-2d5b-44ed-9b2e-882644d7a4ee)